From 50d0f71441a30f7255303b1c4502c9b3ac88dab3 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 25 Nov 2025 09:41:34 +0100
Subject: [PATCH] arm64: dts: st: prevent UART RX DMA channel data packing on
 stm32mp211

To avoid potential byte loss when stopping the UART RX DMA channel, prevent
data packing. Memory accesses will be penalized, but currently, this is the
only way to prevent byte loss.

Change-Id: I95ea63697cb572d9c6e142cba61f435cdaf94a2d
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/512805
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -779,7 +779,7 @@
 				reg = <0x400e0000 0x0 0x400>;
 				interrupts-extended = <&exti1 27 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
-				dmas = <&hpdma 11 0x40 0x12>,
+				dmas = <&hpdma 11 0x40 0x10012>,
 				       <&hpdma 12 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 32>;
@@ -793,7 +793,7 @@
 				reg = <0x400f0000 0x0 0x400>;
 				interrupts-extended = <&exti1 28 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
-				dmas = <&hpdma 13 0x40 0x12>,
+				dmas = <&hpdma 13 0x40 0x10012>,
 				       <&hpdma 14 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 33>;
@@ -807,7 +807,7 @@
 				reg = <0x40100000 0x0 0x400>;
 				interrupts-extended = <&exti1 30 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
-				dmas = <&hpdma 15 0x40 0x12>,
+				dmas = <&hpdma 15 0x40 0x10012>,
 				       <&hpdma 16 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 34>;
@@ -821,7 +821,7 @@
 				reg = <0x40110000 0x0 0x400>;
 				interrupts-extended = <&exti1 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
-				dmas = <&hpdma 17 0x40 0x12>,
+				dmas = <&hpdma 17 0x40 0x10012>,
 				       <&hpdma 18 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 35>;
@@ -1027,7 +1027,7 @@
 				reg = <0x40220000 0x0 0x400>;
 				interrupts-extended = <&exti1 29 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
-				dmas = <&hpdma 19 0x40 0x12>,
+				dmas = <&hpdma 19 0x40 0x10012>,
 				       <&hpdma 20 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 36>;
@@ -1436,7 +1436,7 @@
 				reg = <0x40370000 0x0 0x400>;
 				interrupts-extended = <&exti1 32 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
-				dmas = <&hpdma 21 0x40 0x12>,
+				dmas = <&hpdma 21 0x40 0x10012>,
 				       <&hpdma 22 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 37>;
